src/utils.js

```diff
@@ -27,23 +27,47 @@ const getAccessToken = () => {
   return cookies.get('access_token');
 };
 
-const updateUrl = (data) => {
-  if (!data) {
+const updateUrl = (queryOptions) => {
+  if (!queryOptions) {
     return;
   }
   const currentQuery = qs.parse(window.location.search);
-  const newQuery = qs.stringify({
+
+  // Apply any updates passed in over the current query. This requires consumers to explicitly
+  // pass in parameters they want to remove, such as resetting the page when sorting, but ensures
+  // that we bring forward all other params such as feature flags
+  const newQuery = {
     ...currentQuery,
-    page: data.page !== 1 ? data.page : undefined,
-    ordering: data.ordering,
-    search: data.search,
-  });
+    ...queryOptions,
+  };
+
+  // Because we show page 1 by default, theres no reason to set the url to page=1
+  if (newQuery.page === 1) {
+    newQuery.page = undefined;
+  }
 
-  if (newQuery !== window.location.search) {
-    history.push(`?${newQuery}`);
+  const newQueryString = `?${qs.stringify(newQuery)}`;
+  if (newQueryString !== window.location.search) {
+    history.push(newQueryString);
   }
 };
 
+const getPageOptionsFromQS = () => {
```

Comment:
nit: It might be worthwhile to add a comment above describing this function to make it clearer what its purpose is and that the `QS` in the function name refers to getting options from the URL query string.